### PR TITLE
Add support for >= Python 3.8

### PR DESCRIPTION
**Pull request by micat001**
# Checklist

- [X] I have performed a thorough review of my code
- [X] I have sufficiently commented my code
- [X] The implementation follows the project style conventions
- [X] All project unit tests are passing
- [X] If relevant, documentation has been provided or updated to discuss the changes made
- [X] System integration tests were performed successfully

## Changes Made

Change to python version required. Used [vermin](https://github.com/netromdk/vermin) to check for feature compatibility. 

## Associated Issues

N/A

## Files Changed

`pyproject.toml`: Changed the required python version to 3.8 from 3.10.
* Used [vermin](https://github.com/netromdk/vermin) to check the required python version. Minimum required is 3.7. 
* Almost everything we are using to test is 3.9, so instead of needlessly upgrading python version, just downgraded the required version of this package.
* Per @amarburg suggestion, setting the minimum version to be 3.8 (this is because 3.8 is default on 20.04 ubuntu and with ros noetic)

## Testing
Tested using the pytest scripts. 
Have also ran in-person on the Bravo Arm at UW-APL.
Ran example files to check proper functionality.

### CODE REVIEW

**Comment by micat001** [comment]
Hey @evan-palmer!

I am working at APL with @amarburg and company. We have been using the python driver for the bravo arm routinely the last few weeks and I'm really appreciating the work you've put in to get it running in Python. Opening the MR to address something both Aaron and I changed about our local install. 

Thanks!

File: pyproject.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ authors = [
   { name = 'Evan Palmer', email='evanp922@gmail.com' },
 ]
 license = {file = 'LICENSE'}
-requires-python = '>=3.10'
+requires-python = '>=3.7'
```
Comment (evan-palmer) [inline pyproject.toml:14]: Have you had any issues with the type-hints used? The current type-hints target <= Python 3.9 which I have seen cause  problems in the past when using Python 3.7. I don't anticipate any build or run-time issues, but want to make sure that this doesn't introduce any issues

Comment (micat001) [reply pyproject.toml:14]: I created a couple of empty venvs to test this -

We mostly have python 3.8 and 3.9 running, but I also created a 3.7 version.  In that case, there were compatibility issues with the `crc` and `cobs` library minimum versions, so i bumped those down as well.

I added a `from __future__ import annotations` to the `driver.py` to (hopefully) avoid any typing issues. I was able to run both examples and the `test_protocol.py' in a python 3.7, 3.8, and 3.9 environment.

All this being said, if you're more comfortable just bumping the minimum down to 3.9, that seems groovy too.

Comment (evan-palmer) [reply pyproject.toml:14]: In that case, let's go ahead and drop down the minimum `crc` and `cobs` versions and integrate the `__future__` module to support Python 3.7

Comment (amarburg) [reply pyproject.toml:14]: I'd like to put in a vote for 3.8 as it's default version for Ubuntu 20.04 (for those of us stuck with Noetic)

Comment (micat001) [reply pyproject.toml:14]: @amarburg  - That's a good point and a good reason to target 3.8 over 3.7.
If targeting 3.8, the crc and cobs version can stay where they were (seems to install fine in a 3.8 env)

Comment (micat001) [reply pyproject.toml:14]: I can push that change (change cobs/crc versions back, change py min to 3.8) if @evan-palmer is okay with 3.8 as the minimum?

Comment (evan-palmer) [reply pyproject.toml:14]: 3.8 should be fine. The current dependency minimum versions should be able to stay the same as @micat001 has said (see [here](https://github.com/Nicoretti/crc/blob/1e43c81e232ed0cfc2077e0d72414c7d230c8ba5/pyproject.toml#L29) and [here](https://github.com/cmcqueen/cobs-python/blob/main/setup.py)). The future module will still need to be included to support the >=Python 3.9 type hints though

Comment (micat001) [reply pyproject.toml]: reverted to 3.8 and tested on a clean install. I've also fixed a small bug in `send_joint_commands.py` with the enum as i was using that file for tested.

**Review by evan-palmer — APPROVED** [review]
Looks good! Thanks for contributing!

**Pull request merged**